### PR TITLE
JS: support module.createRequire

### PR DESCRIPTION
**Pull request by erik-krogh**
Documentation: https://nodejs.org/api/modules.html#modules_module_createrequire_filename

Files that use the ES2015 module system does not have access to the `require()` function.   
However, you can create an instance of `require` in the following way: 

```JavaScript
import {createRequire} from "module";
let require = createRequire(import.meta.url)
```

With this PR we support that pattern.   
[Here is a file that use the pattern.](https://github.com/conermurphy/covid-19-graphql-api/blob/master/index.js). 

`createRequire(path)` does take a path as argument, and this path determines from where future imports happen.  
However, that path is often either very difficult to determine, or the path is `import.meta.url` (a path to self).   
So in the vast majority of cases we can safely ignore that path.  
(And even if we get it wrong, it is unlikely that we import the wrong file, we will likely just be unable to resolve the import). 

I encountered this while looking at how the module systems work.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/NodeJS.qll [hunk 1]
```diff
@@ -162,6 +162,8 @@ private predicate isRequire(DataFlow::Node nd) {
   not nd.getFile().getExtension() = "mjs"
   or
   isRequire(nd.getAPredecessor())
+  or
```
Comment (esbena) [inline javascript/ql/src/semmle/javascript/NodeJS.qll:165]: Do we dare to use `getACall` here? Shouldn't this be on the more syntactic side, and let the above `isRequire(nd.getAPredecessor())` take care of the data flow?

Comment (erik-krogh) [reply javascript/ql/src/semmle/javascript/NodeJS.qll:165]: `getACall` only handles data-flow for the `require("module").createRequire` property.  
Data-flow for the result of that call is completely handled  by `isRequire(nd.getAPredecessor())`, there is no overlap. 

I could use the below check instead to make it more syntactic (and miss some flow), but I still want to use `DataFlow::moduleMember()` to support various kinds of imports (destructing etc.). 

```CodeQL
exists(CallExpr call |
  call.getCallee() = DataFlow::moduleMember("module", "createRequire").asExpr()
|
  nd.asExpr() = call
)
```

Comment (esbena) [reply javascript/ql/src/semmle/javascript/NodeJS.qll:165]: Right. My issue is actually with the recursive use of `moduleMember`, sorry for the `getACall` red herring. I am happy to see it merged as is, but I believe we historically have struggled with the performance/polarity of recursive `moduleImport` predicates. Perhaps I am wrong, @asgerf do you remember?

**Comment by asgerf** [comment]
@erik-krogh Can you run a small evaluation?

**Pull request merged**